### ISSUE TITLE
Elusive Nans

### ISSUE THREAD

**Issue by Hamilton232** [MIC-DKFZ/nnUNet#206]
Hi Thank you so much for the terrific code.

So I converted my dataset into your format but for some reason I receive NaNs for the training and validation loss. I decided to create a fake dataset that I know exactly what is in it for testing purposes and I still receive NaNs. 

Any insight would be greatly appreciated!

(Ps. I have tested the code with a MSD dataset and it works fine so there must be something wrong with my conversion code)

----- Fake Dataset Creation Code 

import SimpleITK as sitk
from os.path import exists, isfile, join
from os import listdir, makedirs
from collections import OrderedDict
import random
import json
import numpy as np

makedirs('Task102_Fake/imagesTr')
makedirs('Task102_Fake/labelsTr')
makedirs('Task102_Fake/imagesTs')

train_pct = 1
test_pct = 0
train_ids = []
test_ids = []
for i in range(20):
    roll = random.random()
    #id = str(i)
    id = 'FAKE_' + str(i)
    img_np = np.random.randint(0, 255, size=(1, 10, 10))
    img = sitk.GetImageFromArray(img_np)
    label_np = np.random.randint(0, 2, size=(1, 10, 10))
    label = sitk.GetImageFromArray(label_np)

    if roll < train_pct:
        train_ids.append(id)
        sitk.WriteImage(img, join('Task102_Fake/imagesTr', id + "_0000.nii.gz"))
        sitk.WriteImage(label, join('Task102_Fake/labelsTr', id + ".nii.gz"))
    elif roll < train_pct + test_pct:
        sitk.WriteImage(img, join('Task102_Fake/imagesTs', id + "_0000.nii.gz"))
        #sitk.WriteImage(label, join('Task102_Fake/labelsTr', id + ".nii.gz"))

json_dict = OrderedDict()
json_dict['name'] = "LITS"
json_dict['description'] = "LITS"
json_dict['tensorImageSize'] = "4D"
json_dict['reference'] = "see challenge website"
json_dict['licence'] = "see challenge website"
json_dict['release'] = "0.0"
json_dict['modality'] = {
    "0": "FakeImage"
}

json_dict['labels'] = {
    "0": "background",
    "1": "label"
}

json_dict['numTraining'] = len(train_ids)
json_dict['numTest'] = len(test_ids)
json_dict['training'] = [{'image': "./imagesTr/%s.nii.gz" % i, "label": "./labelsTr/%s.nii.gz" % i} for i in train_ids]
json_dict['test'] = ["./imagesTs/%s.nii.gz" % i for i in test_ids]

with open(join('Task102_Fake', "dataset.json"), 'w') as f:
    json.dump(json_dict, f, indent=4, sort_keys=True)

**Comment by Rebecca3996** [comment]
Hi, 
I am facing the exact same issue. I haven't resolved it yet. But you can crosscheck your code with this:
https://github.com/MIC-DKFZ/nnUNet/issues/172#issuecomment-621145412

**Comment by FabianIsensee** [comment]
Hi, thanks for raising this issue and especially for including code with which to reproduce it. I will look into this!
Best,
Fabian

**Comment by FabianIsensee** [comment]
Hi,
I found a bug that would give the loss a nan weight if only one segmentation output was configured (no auxiliary outputs). This is now fixed (you need to pull the current code from master)

This is what your dummy task now looks like for me:

> 2020-04-30 13:40:20.480796: 
> epoch:  3
> 2020-04-30 13:40:23.959372: train loss : 0.0277
> 2020-04-30 13:40:24.259872: validation loss: 0.5577
> 2020-04-30 13:40:24.260122: Average global foreground Dice: [0.5319229316207027]
> 2020-04-30 13:40:24.260170: (interpret this as an estimate for the Dice of the different classes. This is not exact.)
> 2020-04-30 13:40:24.992400: lr: 0.009964
> 2020-04-30 13:40:24.992511: This epoch took 3.779127 s
> 

I don't think this issue is related to @Rebecca3996 's because she is working on KiTS data and the problem here was caused by the extremely small image size.
Best,
Fabian

**Comment by Hamilton232** [comment]
Thanks for the super quick response! You rock!

I will update and check it out!
